### ISSUE TITLE
How does percent_reduced reduce the dataset?

### ISSUE THREAD

**Issue by Kerro-junior** [gnina/scripts#50]
When I looked into the train.py script, I found this parameter a little mysterious.

### use_reduced = bool(args.reduced or args.percent_reduced)
According to this code, it should be a parameter to reduce the sample of the training set and test set(or not?).

### Early stopping criteria
According to your paper "Early stopping evaluates the loss of the trained network every 1000 iterations on a sample of the training set. The size of this sample is determined by the percent_reduced parameter to [train.py].", this percent_reduced is something about the training set but also not sure whether related to test set. And I wonder if I set percent_reduced to 5, what number of Early Stopping will be?

### ChatGPT ANSWER
According to ChatGPT, the percent_reduced is something about the learning_rate: "In gnina, the percent_reduced parameter does not directly reduce the dataset. My earlier answer was inaccurate. I apologize for the confusion.
The percent_reduced parameter in gnina's [train.py](http://train.py/) script is used to specify the percentage of reduced learning rate schedule to be applied during training. It controls the rate at which the learning rate is reduced over time.
The reduction of dataset samples can be achieved through other means, such as data preprocessing steps or by providing a subset of the original dataset to the train.py script using other command-line arguments or data loading techniques. "
SO, is its answer totally wrong?

**Comment by JonasLi-19** [comment]
parser.add_argument('--percent_reduced',type=float,default=0,help='Create a reduced set on the fly based on types file, using the given percentage: to use 10 percent pass 10. Range (0,100). Incompatible with --reduced')

**Comment by dkoes** [comment]
In order to determine when to stop training, we want to evaluate on the entire training set (could also do a validation set, but that's not what we do by default).  However, the CrossDocked set is very large and if we have to evaluate the entire set every time then training takes too long, so we evaluate a reduced subset.

**Comment by francoep** [comment]
The chatGPT answer is nonsense.

With CrossDocked the dataset is very large as David said. So, we desired a smaller subset to perform our evaluations. You can specify that manually with --reduced   OR use --percent_reduced to have the code generate a reduced set for you (where passing --percent_reduced 10  => means 10% of the datasets will be converted to a reduced set).

use_reduced is a flag that tells the training script if the evaluations require a reduced set or not.

Early stopping is controlled by three entirely different parameters -->  --step_when  & step_end_cnt  &  -t.

-t sets the number of iterations that pass before you do your evaluations. We used -t 1000.
--step_when  determines the number of evaluations that need to pass with no improvement in the training loss for the learning rate to lower (by a factor of 10).
--step_end_cnt   determines the number of times the learning rate is allowed to lower.

So -t 1000 --step_when 200 --step_end_cnt 3  ==>   every 1000 iterations the training set is evaluated (or reduced training set)  if performance improves we continue and mark the current iteration count as the "best". If not, we check if there was no improvement in the last 200 checks since the "best". If we are under 200 checks, we continue. Otherwise, we attempt to lower the learning rate. If we are allowed to (hasn't occured step_end_cnt times) we lower the learning rate and continue training. Otherwise training stops.
